### ISSUE TITLE
Rewrite 0.7 client support in the server to reside completely outside of the main code

### ISSUE THREAD

**Issue by heinrich5991** [ddnet/ddnet#7779]
Messages would get translated in extra files, the main game code would not need to concern itself with that.

Ideally, this could also handle old DDNet clients.

Would me implementing such a solution be acceptable? An upgrade over the status quo? @Robyt3 @ChillerDragon @edg-l @Jupeyy @Learath2 ?

**Comment by Jupeyy** [comment]
If you ask me, we should rather look into making a better protocol, easier to use, less hacky, than being compatible to a unmaintained project.

If you want to waste your time for 0.7 go ahead, if it's completely decoupled from ddnet, it can at least easily be made optional or be removed.

**Comment by heinrich5991** [comment]
> If you ask me, we should rather look into making a better protocol, easier to use, less hacky, than being compatible to a unmaintained project.

I'm doing that at the same time for the low-level protocol, see #6961. Do you mean for the high-level DDNet protocol we currently use?

**Comment by Learath2** [comment]
A million times, yes. The more I implemented this current solution, the more I hated it. I just couldn't figure out how to do it without wasting a lot of processing power.

Issue is the more you isolate the translation layer, the more you need to re-derive as information is lost. Maybe  one could use more hooks than just at the packet layer.

**Comment by Learath2** [comment]
I also do kinda agree with @Jupeyy, though. With how dead 0.7 is, I wonder if you would be wasting your precious time on something a dozen people would use.

**Comment by Jupeyy** [comment]
> I'm doing that at the same time for the low-level protocol, see #6961. Do you mean for the high-level DDNet protocol we currently use?

Yes, i think working with this Unpacker class is a pain. Additionally it feels like too many packets are seperated in different packets to store extra information, this isn't intuitive neither for new contributors nor for me.

E.g. if i want to extend the laser to a have a owner id, i'd simply add that to a laser snapshot, not some entityex, or whatever

I simply dislike this compability thinking that acts like software must be perfect, and if it's not we build work arounds.
Work arounds are never intuitive, and tend to break, because you can easily oversee them.

As for the low level protocol, I cannot judge about what your plans are, but it should be threaded, and it should be more dynamic. Not being able to send all rcon commands at once bcs of some weird internal limitation is very weird (i dunno if that's the case, but sounded like it when we had this rcon command pr open)

**Comment by ChillerDragon** [comment]
I have nothing against it. Not sure if it is much better. And I also would like to see your time used otherwise. https://github.com/ddnet/ddnet/pull/4791 for example is much cooler!

**Comment by jxsl13** [comment]
would be pretty cool
